### PR TITLE
Bug fix for issue #1289

### PR DESCRIPTION
**Pull request by sshu88**
Fix the issue of no wood product when forced by area-based harvest.  https://github.com/NGEET/fates/issues/1289

### Checklist

*If this is your first time contributing, please read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/main/CONTRIBUTING.md) document.*

All checklist items must be checked to enable merging this pull request:

*Contributor*
- [x] The in-code documentation has been updated with descriptive comments
- [x] The documentation has been assessed to determine if updates are necessary

*Integrator*
- [x] FATES PASS/FAIL regression tests were run
- [x] Evaluation of test results for answer changes was performed and results provided

### Documentation

- [Technical Note](https://github.com/NGEET/fates-docs) update:
- [User's Guide](https://github.com/NGEET/fates-users-guide) update: 

### Test Results:

*CTSM (or) E3SM (specify which) test hash-tag:*

*CTSM (or) E3SM (specify which) baseline hash-tag:*

*FATES baseline hash-tag:*

*Test Output:*

### CODE REVIEW

**Review by XiulinGao — APPROVED** [review]
This looks good to me!

File: biogeochem/EDLoggingMortalityMod.F90 [hunk 1]
```diff
@@ -291,7 +291,7 @@ subroutine LoggingMortality_frac( pft_i, dbh, canopy_layer, lmort_direct, &
          ! transfer of area to secondary land is based on overall area affected, not just logged crown area
          ! l_degrad accounts for the affected area between logged crowns
          if(prt_params%woody(pft_i) == itrue)then ! only set logging rates for trees
-            if (cur_harvest_tag == 0) then
+            if (cur_harvest_tag == 0 .or. cur_harvest_tag == 2) then
```
Comment (ckoven) [inline biogeochem/EDLoggingMortalityMod.F90:294]: HI @sshu88  -- would it be possible to replace these integers that you are comparing `cur_harvest_tag ` to, here and elsewhere, with named integer constants that describe more clearly what these conditions refer to?

Comment (sshu88) [reply biogeochem/EDLoggingMortalityMod.F90]: Nice suggestion! I updated the code.

**Comment by glemieux** [comment]
Regression testing underway on derecho.

**Comment by glemieux** [comment]
Testing the `fates_cold_landuse` testmod via e3sm on `perlmutter`  exposed unrelated issue e3sm-project/e3sm#6806.  Fixing this via https://github.com/E3SM-Project/E3SM/pull/6762/commits/f7267be17a32d5e1c832df06b78f9cbdbd15f9bb results in the e3sm landuse test passing.

Results: `/global/homes/g/glemieux/scratch/e3sm-tests/pr1291-landuse-diag3-6806fix.fates.pm-cpu.gnu.Ca714d58227-F77e89489`

**Pull request merged**